src/runtime/particle-execution-host.ts

```diff
@@ -96,14 +96,14 @@ export class ParticleExecutionHost {
       async onHandleStore(handle: StorageProviderBase, callback: number, data: {value: {}, keys: string[]}, particleId: string) {
         // TODO(shans): fix typing once we have types for Singleton/Collection/etc
         // tslint:disable-next-line: no-any
-        await (handle as any).store(data.value, data.keys, particleId);
+        await (handle as CollectionStorageProvider).store(data.value, data.keys, particleId);
```

Comment:
It looks like currently this is only used for Collections, I'm not sure whether there was thinking to go beyond that at some point, would need input from @shans

Reply:
I think we can move forward with this and revisit as the new storage API takes shape.